Image after upload is different than one originally photographed
Occasionally, after upload, an image will show in Hydra as the wrong image (usually a duplicate of another image). The filename, thumbnail, and image (after re-download) does not match the original image that was digitized. Here's an example:  - image downloads and shows as 2012.017_b2f38_002.tif but is actually not the original 2012.017_b2f38_002.tif that Jay shot. (Correct record:  I think the earliest Hillary noticed this was with these two images, first ingested around 4/18/2017:   Also happened with these two:  
this happened to  with a file in this work on 7/5:   incorrect file was replaced
My understanding is: * Multiple image files are uploaded together * After upload, one of the files is messed up in the following way   * It has the correct/expected filename   * It has the thumbnail of one of the other files in the same upload batch   * Its file download matches the thumbnail (i.e. is the wrong file)
I asked about this on hydra slack and it doesn't seem like anyone else has seen it. I've also just sent an email to the hydra-tech list and will update if I get any responses. I believe that is what we discussed as our next step for this issue.
I've gotten a couple of leads but none of them look exactly like our bug. Some things to look at more closely:   (look at logs for `WildcardExceptionMapper` error)    
Looked into the leads above. The first two are about a bug which duplicates files, as opposed to replading. But I did check catalina logs (which go back to may) and found no trace of either the `WildcardExceptionMapper` error nor any mention of `hasMimeType`.  The last issue above is related to async jobs, results in files with derivatives not fully processed. Not related to this.  My plan as a next step, since this cannot be replicated, is to trace the path through code from a file being added to a form for upload through that file being deposited in fedora (pretty sure that happens before derivative generation). The bug is somewhere in there and increased familiarity may help identify potential problem areas.
Traced code for a little while and found what looked like a bug in sufia where a carrierwave method was overridden in a way that looked unintentional.  Here's the [override in sufia]( and the [overridden code in carrierwave]( CarrierWave expects `cache_path` to provide the full path + filename of the cached file. But the overwrite makes it give just a base path.  Checked in hyrax to see whether anyone has had noticed and indeed it [has been fixed there]( It was fixed as part of a PR which pulled in recent commits from sufia, during the time those two codebases were being kept in sync. Unfortunately this bugfix commit was not pulled back into sufia.  Poked around some of this code in a debugger for a bit to see whether I could figure out a way it might cause the bug discussed in this issue. I couldn't figure out a concrete explanation, but the overwrite definitely impacts the naming / location of the cached upload file, so it seems possible that it is the source of our problem here.  I'll create a PR to pull in the bug fix from hyrax.  I also plan to step through some of sufia's `app/jobs/attach_files_to_work_job.rb` in case there's anything I can figure out there, where the uploaded file is being saved to fedora. However, I'm inclined to consider the forthcoming overwrite fix sufficient to close this issue unless/until someone sees the bug again. I realize that's not ideal but since we can't replicate there is no way to confirm that any particular action has resolved the bug.
ðŸŒˆ 
Only other area I found where it looked like something could potentially go wrong was  since there's just a bunch of juggling of ids. But I don't think there are bugs there, and I don't see significant changes in hyrax so I think that's not likely to be the problem.  I'm going to do the PR mentioned in the previous comment and let that close this issue unless/until someone experiences the bug again and reopens it.
Re-opening because I just found another instance of this:   Fileset attached to this work isn't the right one. This was a recent batch upload on 9/15/17.